### PR TITLE
Support substring match for --verify-function, fix --verify-root problem

### PR DESCRIPTION
**Pull request by jialin-li**
support unique substring match, beginning and ending wildcard matches for `--verify-function`
fixes the `--verify-root` bug where `--verify-root` will include both the root module and vstd's root module.

### CODE REVIEW

File: source/rust_verify/src/verifier.rs [hunk 1]
```diff
@@ -913,14 +930,21 @@ impl Verifier {
             funs.insert(function.x.name.clone(), (function.clone(), vis_abs));
         }
 
+        let mut exact_match = false;
```
Comment (jaybosamiya) [inline source/rust_verify/src/verifier.rs:933]: Might be good to call this `verify_function_exact_match` or similar?

File: source/rust_verify/src/verifier.rs [hunk 2]
```diff
@@ -934,6 +958,55 @@ impl Verifier {
...
+                let right_wildcard = verify_function.ends_with("*");
+                let mut wildcard_mismatch = false;
+
+                if left_wildcard && !right_wildcard {
+                    wildcard_mismatch =
+                        !filtered_functions.iter().any(|f| f.ends_with(clean_verify_function));
+                } else if !left_wildcard && right_wildcard {
+                    wildcard_mismatch =
+                        !filtered_functions.iter().any(|f| f.starts_with(clean_verify_function));
+                }
```
Comment (jaybosamiya) [inline source/rust_verify/src/verifier.rs:992]: Alternative way of writing this code that eliminates the `let mut`
```suggestion
                let wildcard_mismatch = match (verify_function.starts_with("*"), verify_function.ends_with("*")) {
                  (true, false) => !filtered_functions.iter().any(|f| f.ends_with(clean_verify_function)),
                  (false, true) => !filtered_functions.iter().any(|f| f.starts_with(clean_verify_function)),
                  _ => false,
                };
```

You can also instead do it as a `let wildcard_mismatch = if ...` if you prefer that

**Review by jaybosamiya — APPROVED** [review]
Thanks!  This PR makes `--verify-function` a lot easier to work with for users.

Some minor comments inline.

Also, we probably want to update the `verus --help` message which currently says:
```console
$ verus --help
...
        --verify-function MODULE
                        Verify just one function (e.g. 'foo' or 'foo::bar')
                        within the one module specified by verify-module or
                        verify-root
...
```

File: source/rust_verify/src/verifier.rs (hunk 1 above)
Comment (jialin-li) [reply source/rust_verify/src/verifier.rs]: Both addressed in comments below!

**Review by utaal — APPROVED** [review]
Thank you!

I pushed a commit with a couple of fixes for issues that were already present, but that I only noticed when giving this a try.

**Pull request merged**